Faster Implementation of Wildcard Matching and added test file for wildcard matching

bhautikpip opened the pull request:
*Issue #, if available:*

*Description of changes:*

1. Faster Implementation of Wildcard matching 
2. Added test file for wildcard_match function

I have tested this change with the newly added test function.


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

chanchiem commented:
It looks like code coverage decreased. Can we make sure all code paths are executed in your tests? Additionally, it would be nice if we could get a high level description of the algorithm in the comments for readability purposes.

And can we also confirm that these tests still pass before this change and after the change?

bhautikpip commented:
Tests have passed in both the version before this change and after this change. Moreover, I have added description of the algorithm. Initially code coverage was decreased because in the older version we are using simple_wildcard_change which we do not require in this version so I have removed that function.

chanchiem commented on aws_xray_sdk/core/utils/search_pattern.py:
```diff
@@ -12,74 +12,47 @@ def wildcard_match(pattern, text, case_insensitive=True):
-        pattern = pattern.lower()
-        text = text.lower()
+    # If elif logic Checking different conditions like match between the first i chars in text
+    # and the first j chars in pattern, checking pattern has '?' or '*' also check for case_insensitivity
```
there's no j; it should be a p

bhautikpip replied:
Thanks for the correction! Committed that change.
